### PR TITLE
Revert "Merge pull request #34055 from palegre-tiny/groupSortedArray"

### PR DESCRIPTION
**Pull request by alexey-milovidov**
### Changelog category (leave one):
- Bug Fix (user-visible misbehaviour in official stable or prestable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Remove function `groupArraySorted` which has a bug.

This reverts commit f055d7b692f977155f8c2abc3d3753a89ae0a7c7, reversing
changes made to 4ec3c35e14b5b7ec4005429a1f962155166637ec.

Bug has been found.
The function will not be returned back unless both of these conditions will be satistied:
1. Bug will be fixed #36815.
2. Fuzz testing will be improved to automatically find all these bugs before merge.

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
@mergifyio update

**Comment by azat** [comment]
@alexey-milovidov maybe it worth give this function a try, after #36815?

**Comment by alexey-milovidov** [comment]
It's not the first time we forgot to test ser/de or merge, let's do fuzzing of table engines (replace every table to distributed table).

**Comment by alexey-milovidov** [comment]
Btw, why this particular function useful?

**Comment by azat** [comment]
>Btw, why this particular function useful?

I don't need it, but I guess original developer (@palegre-tiny) will not be happy if it will be removed.
That said that sometimes reverts are too quick, it is great for the code base (I guess), but not for users.
Also this function hadn't been marked as experimental, so someone can use it.

_P.S. I don't have any strong opinion here, this is just basic thoughts about almost every "function"_

**Comment by alexey-milovidov** [comment]
Ok, let's merge the fix first.

**Comment by alexey-milovidov** [comment]
But there are other code cleanups required, maybe you can make full code review and cleanup of this function?

**Comment by UnamedRus** [comment]
> Btw, why this particular function useful?

Because it should allow you to get top x values by sorting.
And doing arraySort(groupArray(xxx)) is costly:
1. how array functions works in memory.
2. in case of distributed query the whose groupArray state will be send to the init node.

**Comment by palegre-tiny** [comment]
> 

That is. We need this feature for some scenarios and we didn't see any acceptable alternatives. I could clean up or simplify the code if it's not clear.

**Comment by alexey-milovidov** [comment]
Do you mind if I will revert it *temporarily* just because it has memory safety issue and our master branch must be 100% memory safe? Then we will merge #36815 when it will be ready.

**Comment by palegre-tiny** [comment]
> Do you mind if I will revert it _temporarily_ just because it has memory safety issue and our master branch must be 100% memory safe? Then we will merge #36815 when it will be ready.

Sure, lets do like that. Thank you.

File: tests/queries/0_stateless/02233_optimize_aggregation_in_order_prefix_with_merge.reference [hunk 1]
```diff
@@ -1,69 +1,5 @@
...
-[1,2]	20	200	4000
-
-[1,1,2,2]	0	0	6000
-select groupArraySorted(partition), parent_key, child_key, sum(value) from data_02233 group by parent_key, child_key with totals order by parent_key, child_key settings optimize_aggregation_in_order=1, max_block_size=1;
-[1,2]	10	100	2000
-[1,2]	20	200	4000
-
-[1,1,2,2]	0	0	6000
--- sum() can be compiled, check that compiled version works correctly
-select groupArraySorted(partition), parent_key, child_key, sum(value) from data_02233 group by parent_key, child_key with totals order by parent_key, child_key settings optimize_aggregation_in_order=1, compile_aggregate_expressions=1, min_count_to_compile_aggregate_expression=0;
```
Comment (azat) [inline tests/queries/0_stateless/02233_optimize_aggregation_in_order_prefix_with_merge.reference:52]: Now we don't have explicit test for compiled functions...
Will address this later.

**Pull request merged**